### PR TITLE
feat: add type in schema Port to support multiple clouds

### PR DESCRIPTION
**Pull request by healthjyk**
## What type of PR is this?

/kind feature

## What this PR does / why we need it:

 add type in schema Port to support multiple clouds

### CODE REVIEW

File: models/schema/v1/monitoring/prometheus.k [hunk 1]
```diff
@@ -49,4 +49,4 @@ schema Prometheus:
     check:
         regex.match(interval, r"[A-Za-z0-9_.-]*"), "a valid interval must starts with alphanumeric and end with y/w/d/h/m/s/ms"
         regex.match(timeout, r"[A-Za-z0-9_.-]*"), "a valid timeout must starts with alphanumeric and end with y/w/d/h/m/s/ms"
-        port and path if operatorMode == False, "port and path must be set only when operator mode is false"
\ No newline at end of file
+        # port and path if operatorMode == False, "port and path must be set only when operator mode is false"
```
Comment (SparkYuan) [inline models/schema/v1/monitoring/prometheus.k:52]: Why do you comment on this line?

File: models/schema/v1/workload/network/port.k [hunk 2]
```diff
@@ -37,7 +43,7 @@ schema Port:
     # Public defines whether to expose the port through Internet.
     public:                     bool = False
 
-
     check:
+        type if public, "type must exist when public is True, inclusive"
```
Comment (SparkYuan) [inline models/schema/v1/workload/network/port.k:47]: Type is inclusive?

File: models/schema/v1/monitoring/prometheus.k (hunk 1 above)
Comment (healthjyk) [reply models/schema/v1/monitoring/prometheus.k:52]: there is no operatorMode field

File: models/schema/v1/workload/network/port.k (hunk 2 above)
Comment (healthjyk) [reply models/schema/v1/workload/network/port.k:47]: I'll delete it

**Pull request merged**